### ISSUE TITLE
very few reads processed

### ISSUE THREAD

**Issue by phil-grayson** [pachterlab/kallisto#189]
Hi, 

I've spent some time troubleshooting this issue with my data and I think I've got it sorted out.  I was having an issue where very few reads (variable but often under 100) were being processed during kallisto quant despite having 10s of millions of reads in each dataset.  After digging through the fastq files, I discovered that there were a few empty reads that looked something like this:
```
@HEADER:INFO

+

```
It seems that whenever kallisto encounters one of these reads (in paired or single-end mode), it stops.  For me, the solution that seems the simplest and easiest was to replace the empty newlines with a single "N" using the following: `sed 's/^$/N/' file1.fastq > fixed_file1.fastq`
Now the empty reads look like this:
```
@HEADER:INFO
N
+
N
```

In the case of the file that I just tested, it brought my reads processed from 44 to 34,615,469.  

Could a flag be added to the kallisto quant output when it encounters an empty read, or could this issue to fixed in future versions?

Thanks!

**Comment by skurscheid** [comment]
In my view it is good data analysis practice to pre-process FASTQ files prior to analysing with down-stream tools. Something like https://github.com/OpenGene/fastp or FASTQC spring to mind (which also has a switch for removing short sequences).

I personally would never conduct any form of analysis on sequencing data without having performed quality checks, adapter removal etc. It saves a lot of time and grief in the long run.

**Comment by phil-grayson** [comment]
Thanks for the suggestion and the quick reply @skurscheid!  I will definitely check fastp out.  I did check the FASTQC outputs and I utilized an adapter trimmer, but some (like cutadapt - https://cutadapt.readthedocs.io/en/stable/guide.html) leave empty reads by default and I had no problems taking these same files through STAR/RSEM/Limma/Glimma/Voom/EdgeR so it took me a little while to track the problem down.

**Comment by skurscheid** [comment]
No problem! I have honestly not come across empty FASTQ records, but am using the fastp filters to throw out the short ones. Good luck!

**Comment by mschilli87** [comment]
@skurscheid: Have a look at https://github.com/OpenGene/fastp/issues/111. :wink:

**Comment by phil-grayson** [comment]
looks like bcl2fastq2 v2.2 was run to process these reads.

bcl2fastq --adapter-stringency 0.9 --barcode-mismatches 0 --fastq-compression-level 4 --ignore-missing-bcls --ignore-missing-filter --ignore-missing-positions --min-log-level INFO --minimum-trimmed-read-length 0 --sample-sheet SampleSheet.csv --runfolder-dir run_dir --output-dir out_dir --processing-threads 8 --no-lane-splitting --use-bases-mask y38,i8,i8,y38

It seems like perhaps the minimum read length of 0 was the issue here?

**Comment by skurscheid** [comment]
@phil-grayson yes, that switch sure looks like the culprit! Quite a perplexing parameter choice to me. Thanks for getting back!

**Comment by dlaehnemann** [comment]
Have just encountered the same problem. Also, this is a duplicate of issue #135, so this issue here can maybe be closed in favour of that one?
